pkg/controller/deployment/util/deployment_util.go

```diff
@@ -415,14 +415,25 @@ func SetReplicasAnnotations(rs *extensions.ReplicaSet, desiredReplicas, maxRepli
 
 // MaxUnavailable returns the maximum unavailable pods a rolling deployment can take.
 func MaxUnavailable(deployment extensions.Deployment) int32 {
-	if !IsRollingUpdate(&deployment) {
+	if !IsRollingUpdate(&deployment) || *(deployment.Spec.Replicas) == 0 {
```

Comment:
Actually, maybe we should handle this in `MinAvailable` and uses `MinAvailable` instead of `*(deployment.Spec.Replicas)-MaxUnavailable(*deployment)` in those status checks?